packages/core/src/actors.ts

```diff
@@ -274,48 +158,46 @@ export function createDeferredBehavior<TEvent extends EventObject>(
 
       return undefined;
     },
+    subscribe: (observer) => {
+      observers.add(observer);
+
+      return {
+        unsubscribe: () => {
+          observers.delete(observer);
+        }
+      };
+    },
     initialState: undefined
   };
 
   return behavior;
 }
 
-export function createPromiseBehavior<T, TEvent extends EventObject>(
+export function fromPromise<T, TEvent extends EventObject>(
```

Comment:
Somewhat oddly we have a minor inconsistency in the signature (and in other functions here). Those functions here (`from*`) have their type params flipped if we take `Behavior`/`ActorRef` as a point of reference:
```ts
declare function fromReducer<TState, TEvent extends EventObject>(...): ...;
declare function fromCallback<TEvent extends EventObject>(...): ...;
declare function fromPromise<T, TEvent extends EventObject>(...): ...;
declare function fromObservable<T, TEvent extends EventObject>(...): ...;
declare function fromEventObservable<T extends EventObject>(...): ...;
declare function fromMachine<TMachine extends AnyStateMachine>(...): ...;

interface ActorRef<TEvent extends EventObject, TSnapshot = any> { ... }
interface Behavior<TEvent extends EventObject, TSnapshot = any> { ... }
```

Those that take a single type param shouldn't start getting a second one, but I find it somewhat confusing that those that take 2 type params have them flipped.

`fromMachine` is completely at odds here - but it is indeed special here (I don't know how to solve problems that this solves without making it special).

Reply:
We should definitely rework this in a separate PR. I don't anticipate anyone using the generics for these `from*` functions, though.